app/packages/core/src/components/Actions/similar/Warning.tsx

```diff
@@ -1,29 +1,45 @@
-import React from "react";
 import { PopoutSectionTitle, useTheme } from "@fiftyone/components";
+import * as fos from "@fiftyone/state";
+import React from "react";
+import { useRecoilValue } from "recoil";
 import { SORT_BY_SIMILARITY } from "../../../utils/links";
 import { ActionOption } from "../Common";
 
 interface Props {
   hasSimilarityKeys: boolean;
   isImageSearch: boolean;
+  modal: boolean;
 }
 
 const Warning: React.FunctionComponent<Props> = (props) => {
   const theme = useTheme();
+  const hasSelectedSamples = useRecoilValue(fos.hasSelectedSamples);
+  const hasSelectedLabels = useRecoilValue(fos.hasSelectedLabels);
+  const hasSorting = Boolean(useRecoilValue(fos.similarityParameters));
+
+  const isImageSearch =
```

Comment:
It feels like `isImageSearch` and `props.isImageSearch` have the same meaning. Is there a difference? Could a common `selector` be used?

Reply:
Yeah i can make it one selector, as the icon also uses this.